src/acl/RegexData.cc

```diff
@@ -256,7 +251,9 @@ ACLRegexData::parse()
         }
     }
 
-    if (!compileOptimisedREs(data, sl)) {
+    auto options = dynamic_cast<Acl::CaseLineOptions *>(dirtyLineOptions());
+    assert(options);
+    if (!compileOptimisedREs(data, sl, options->isCaseInsensitive())) {
```

Comment:
I assume all these direct dirtyLineOptions() calls are _correct_ (in a sense that they happen on a reset() options). We used the word "dirty" to emphasize that this method should not be called directly. However, the current PR code does call this method directly in several contexts, effectively violating our own defense system and encouraging further violations down the road. If this is the best we can come up with, then we should probably drop the "dirty" suffix.

The problem identified here may have a better (and a completely different) solution, but the specific solution outlined above (i.e. to drop the "dirty" prefix) is compatible with the other change request that suggests using a static class member for these options.